flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/expressions/ExpressionParser.scala

```diff
@@ -59,6 +59,8 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
           Literal(str.toInt)
         } else if (str.endsWith("f") | str.endsWith("F")) {
           Literal(str.toFloat)
+        } else if (str.endsWith(".")) {
```

Comment:
@tillrohrmann 
This '.' related change I think I need more time to understand how this parser works. So for this PR can i only handle the string concat part. I have ensured that 42+a+b or a + b + 42 works fine.